### PR TITLE
Make Essentials lifecycle registration happen by default

### PR DESCRIPTION
**Pull request by Redth**
Currently we have a few samples that are adding their own startup code to register the right lifecycle events and wire them back up to Essentials'.

Ideally this is done for users when using the MauiAppBuilder, and so this PR brings the good ideas from the MAUI Repo sample into Core itself and automatically wires up the necessary essentials lifecycle events so that the consumer doesn't need to do this.

It also brings the concept of ConfigureEssentials where you can add app actions, enable version tracking, set the maps key, etc.

### CODE REVIEW

File: src/Core/src/Hosting/MauiAppBuilder.cs [hunk 1]
```diff
@@ -56,6 +57,8 @@ internal MauiAppBuilder(bool useDefaults = true)
 				this.ConfigureCrossPlatformLifecycleEvents();
 				this.ConfigureDispatching();
 			}
+
+			this.UseEssentials();
```
Comment (Eilon) [inline src/Core/src/Hosting/MauiAppBuilder.cs:61]: Should this go inside `if (useDefaults)`? Just an open question, I have no horses and this isn't a horse race...

**Review by jamesmontemagno — APPROVED** [review]

File: src/Core/src/Hosting/MauiAppBuilder.cs (hunk 1 above)
Comment (Redth) [reply src/Core/src/Hosting/MauiAppBuilder.cs:61]: I had this thought too... Maybe not? but I believe we depend on some things happening here like on android for permissions that some API's will not work if they aren't wired up to the right lifecycle events...

I'm could go either way on this one...

Comment (Eilon) [reply src/Core/src/Hosting/MauiAppBuilder.cs:61]: Yeah I don't have background on this unfortunately. Another way to look at it, is it "more" or "less" needed than what's already in the `if()`? Or about the same?

**Review by jamesmontemagno — APPROVED** [review]

File: src/Core/src/Hosting/MauiAppBuilder.cs (hunk 1 above)
Comment (Redth) [reply src/Core/src/Hosting/MauiAppBuilder.cs:61]: Yea I think I'll move this inside, given that things like Handlers and Dispatching are in there already, this feels about the same level of 'importantness'

File: src/Core/src/Platform/Android/MauiAppCompatActivity.cs [hunk 2]
```diff
@@ -28,6 +29,8 @@ protected override void OnCreate(Bundle? savedInstanceState)
 
 			base.OnCreate(savedInstanceState);
 
+			MauiApplication.Current.Services?.InvokeLifecycleEvents<AndroidLifecycle.OnCreate>(del => del(this, savedInstanceState));
```
[the remark this answers is not in the document]
Comment (Redth) [reply src/Core/src/Platform/Android/MauiAppCompatActivity.cs:32]: Ahh i missed this... but in reality i supect we almost want 2 methods here especially for OnCreate... one at the start, and one at the end... though at the point you need that much control over the order, there's always the subclass...

[the remark this answers is not in the document]
Comment (mattleibow) [reply src/Core/src/Platform/Android/MauiAppCompatActivity.cs:32]: Actually, reading the docs:

> Called when the Activity calls `super.onCreate()`.

So basically, this line is not needed at all and the event should be firing in the exact same place.

[the remark this answers is not in the document]
Comment (Redth) [reply src/Core/src/Platform/Android/MauiAppCompatActivity.cs]: Yep, removed my accidental duplicate.

**Review by mattleibow — APPROVED** [review]

**Review by jamesmontemagno — APPROVED** [review]

**Comment by rmarinho** [comment]
Build is failing 

```
  D:\agent\1\s\src\Core\src\Platform\Windows\MauiWinUIWindow.cs(68,125): error CS1503: Argument 2: cannot convert from 'System.IntPtr' to 'Microsoft.Maui.Platform.WindowsNativeWindowSubclassedEventArgs' [D:\agent\1\s\src\Core\src\Core-net6.csproj]
```

**Pull request merged**